### ISSUE TITLE
Events no longer working

### ISSUE THREAD

**Issue by AlanSmithe** [jvmahon/homebridge-homeseer#82]
Hi jvmahon,
I'm very happy with your plugin and the constant updates it gets.
However I've a problem running events from Homekit.

When I run an event from Homekit I see that Homebrigde logs the event but HomeSeer isn't.
`
[2019-7-7 21:19:10] [HomeSeer] Setting event value to true
Sfeerverlichting: launchEvent function succeeded!
[2019-7-7 21:19:12] [HomeSeer] Sfeerverlichting: Momentary switch reseting to 0
[2019-7-7 21:19:12] [HomeSeer] Setting event value to false
Sfeerverlichting: launchEvent function succeeded!
`

This is my config that has always worked till now:
`
"events": [
    {
        "name": "Sfeerverlichting",
        "eventGroup": "Lighting",
        "eventName": "Lighting - 1st Floor - Chill Mode"
    },
`

When I run the event from a HMTL GET it works
[Homeseer.url/JSON?request=runevent&group=Lighting&name=Lighting%20-%201st%20Floor%20-%20Chill%20Mode](url)

**Comment by jvmahon** [comment]
I find the event mechanism quite awkward and I probably should have removed it from the plugin.

A better way to run HomeSeer events from HomeKit is to first create a virtual binary switch (On / Off) in HomeSeer.  That virtual switch should be set up with 0 as the Off value, and 255 as the On value.  Add that virtual switch to your config.json so it appears in the iOS Home application.   

Then, in HomeSeer set up two events - on that activates when the switch turns on, the other that activates when the switch turns off. Use these to then run your other HomeSeer event.  

You could also set up just a single event in HomeSeer that activates when the switch turns On - use that to then run your other event and as the last step in the event handling, set the virtual switch back to "off" so it is ready to be used again.

This technique is pretty much what I describe at this Wiki entry for Garage doors. (https://github.com/jvmahon/homebridge-homeseer/wiki/Supporting-non--ZWave-Devices-and-Plugin-values-by-Mapping-to-a-Virtual-HomeSeer-Device )

**Comment by AlanSmithe** [comment]
I agree with you that the switch for an event that always turns of is quite unusual.
I use the event to change the input of my Onkyo receiver and have every HDMI port to an event.
The on/off switch won't work for that. 
I've read that Homekit started supporting input select for a couple tv's maybe I'll look into that.

**Comment by jvmahon** [comment]
Alan,
If the "old" event mechanism worked, the new method I described will too as both method create a switch in HomeKit that turns on for only a few seconds but while on, can be used to fire other events. I'm guessing I didn't describe it well enough the first time, so here's some more detail: 

1. I assume you have an event that sets your Onkyo receiver input, Call it "Onkyo Event"

2. Set up the HomeSeer virtual switch as I've described so it appears in HomeKit as an On / Off switch.

3. In HomeSeer, Create an event for that virtual switch with the following pattern:

IF  Virtual Switch Turns ON,
Then  Run Onkyo Event
Then Wait 5 seconds
Then Set Device Virtual Switch to Off

This will then create a switch in HomeKit which, when turned on, will turn itself off after 5 seconds (i.e., it resets itself for the next "On" event). So you could think of it acting more like a push button.

**Comment by jvmahon** [comment]
Also, I'm curious - did you do any type of update that would have caused the event to fail? I.e., did you update the plugin version or anything else - its possible I introduced a bug somewhere at some time, but I'd need a bit more detail to figure that out.

**Comment by AlanSmithe** [comment]
Yes I understood that I had to create virtual switches and that I had to create events to toggle the switch to off every time the scene is run. It's just that it's an extra step.

I don't know when it stopt working. I can see that the last time I ran an event was the 10th of July. I've updated your plugin I think one time between that period (last version now) and I've also updated HomeSeer to 3.0.0.534 this month. 

The reason I suspect the plugin is because the direct URL still works but I could be wrong.
